crates/sui-protocol-config/src/lib.rs

```diff
@@ -1656,7 +1651,16 @@ impl ProtocolConfig {
                         cfg.feature_flags.shared_object_deletion = true;
                     }
                 }
-                32 => cfg.feature_flags.accept_zklogin_in_multisig = true,
+                32 => {
+                    cfg.feature_flags.accept_zklogin_in_multisig = true;
+
+                    // Only enable random beacon on devnet
+                    if chain != Chain::Mainnet && chain != Chain::Testnet {
+                        cfg.feature_flags.narwhal_header_v2 = true;
+                        cfg.feature_flags.random_beacon = true;
+                    }
```

Comment:
Why do we need this change?

Reply:
I made backwards incompatible changes to behavior, without moving the feature flag to latest protocol version it will try to create the random object using sui framework functions that I didn't add yet in the earlier version.